### ISSUE TITLE
Can't install V4A magisk module.

### ISSUE THREAD

**Issue by UtkarshKunwar** [therealahrion/ViPER4Android-FX#15]
I received an update from my Magisk Manager (v18.0) and during the install, it got stuck on decompressing. I rebooted, made sure it wasn't listed in the modules and tried a reinstalled. (I rebooted after every step. Mentioning so I don't have to repeat myself) Well and good. It says installed in the list of modules but the app is not in the drawer. Did an uninstall using the zip but still no luck.
Repeated multiple times and thought something might be wrong with the current release and try to install v2.3 but it still doesn't show up after multiple tries. Trying to install as a user app to avoid breaking banking apps.

Any idea what's causing this?

**Comment by Zackptg5** [comment]
I have no idea. The same thing has happened to me ever since I upgrade to magisk 18. It happens inconsistently and also happened to me with osm0sis's busybox which leads me to believe that this is some kind of magisk or magiskmanager bug and is unrelated to unity. Furthermore, this occurred on older versions of unity too. The really interesting thing is that if you flash a unity mod with debug enabled in config.sh and look at that debug log after the weird stuff happens, you'll see that the installation actually continued fine for a while before magisk manager pooped out.
So long story short, I don't think this is a unity bug but a magisk or magisk manager bug. I'm still looking into it though because you never know :/

**Comment by UtkarshKunwar** [comment]
Thanks for the speedy response!
I tried to check using a root file explorer and I discovered in `/sbin/.magisk/img/ViPER4AndroidFX/system/priv-app/ViPER4AndroidFX/`, that when I try to install the apk, my file explorer states that the same version is installed but on trying to install that apk, I get a "`There was a problem while parsing the package.`" Just to confirm, I tried to "reinstall" other apps in a similar way by navigating to their apks in their installed folders and they work fine (Both in `/system/app/` and `/data/app/`). So there is a problem with the packaged apk probably? I don't know but I'm posting the MD5 of the apk just in case `a004ec128c51c3aa1e6ef439cdb50a6d`.

EDIT : Just checked the repo and the MD5 matches the one in `custom/mat/` but if I copy the apk somewhere else, it can magically parse it now but the installation still fails for some reason.

EDIT 2 : Sorry for so many edits. I went through the code, I reflashed my ROM, unrooted, rooted, but apparently some major changes in Magisk 18 are the culprit after all. My ROM comes with a custom busybox so osm0sis's busybox was not the problem. In the end, installed Magisk 17.3, installed V4A 2.4 and my other usual modules, and then upgraded Magisk afterwards to v18. Weird flex but okay.
